src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -230,6 +235,22 @@ protected int doWork() {
         long nextOpticalDuplicateIndex = this.opticalDuplicateIndexes != null && this.opticalDuplicateIndexes.hasNext() ? this.opticalDuplicateIndexes.next() : NO_SUCH_INDEX;
         long nextDuplicateIndex = (this.duplicateIndexes.hasNext() ? this.duplicateIndexes.next() : NO_SUCH_INDEX);
 
+        // initialize variables for optional representative read tagging
+        CloseableIterator<RepresentativeReadIndexer> representativeReadInterator = null;
+        RepresentativeReadIndexer rni = null;
+        int representativeReadIndexInFile = -1;
```

Comment:
I am using '-1' as the null value for integers. This seems to be the convention in other methods like in ReadEnds:
https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/sam/markduplicates/util/ReadEnds.java 

The purpose was to initialize these integers once before we begin to iterate of over each record in dowork(). These variables will only be assigned to real values for a record if the 'TAG_DUPLICATE_SET_MEMBERS' flag is on. Otherwise, they retain their null value.